### DISCUSSION TITLE
Haxeflixel Audio Bitrate / Quality?

### DISCUSSION THREAD

**Discussion by reflekshun** [HaxeFlixel/flixel#2431] [Q&A]
I inserted an .ogg music file into the game and after testing a build, noticed that the audio quality went down (I'm an audio engineer and I can tell with 100% certainty). Does anyone know if Haxeflixel does it's on re-encoding to save space on audio files? 

Or, does anyone know where I can find any information on Haxeflixel audio quality / formats, or if there's any way to control that in Haxeflixel? Many thanks!

**Comment by RichardBray** [comment]
I've seen someone else mention this before and to my knowledge, I can't think of anywhere this is mentioned. If anything it would be OpenFL/Lime related but I will do some more digging.

**Reply by reflekshun** [reply]
Thanks a lot man, I'll also do some digging and testing of my own, do let us know what you find!

**Reply by RichardBray** [reply]
Hey @reflekshun I've looked around online and can't see much on audio quality dropping. Are you targeting HTML5? There my be some compression happening on the browser level as well 🤷

**Reply by reflekshun** [reply]
Hey thanks for doing some digging, I also came up with nada - here is a quick video comparing between the in game audio quality and the original .ogg quality. I double checked, both gameplay and original music audio are in stereo, so it really is some kind of audio bitrate reduction thing happening. Might have to check in with the makers of haxeflixel engine? Not sure of the best people to ask!

https://www.youtube.com/watch?v=9_ZblVioOg8

(PS don't believe we're using HTML5)

**Reply by RichardBray** [reply]
Yeah, it's definitely compressed slightly. The music sounds really good btw. I'll ping the Discord forums to see if they come up with anything. Might be a case of using .mp3 or .wav instead of .ogg files but we'll see.

**Reply by RichardBray** [reply]
So apparently if you load the audio directly with howler it's possible to avoid the compression.

Here's an example by @01010111 
https://github.com/01010111/discjam/blob/master/src/scenes/Stage.hx#L54

And here's the documentation:
https://www.openfl.org/learn/npm/api/pages/lime/media/howlerjs/Howl.html

It says it supports all platforms but I know howler is a web library so 🤷 worth a shot.

There's also AudioSource from OpenFL which might do a similar thing for native targes?
https://www.openfl.org/learn/npm/api/pages/lime/media/AudioBuffer.html

**Reply by reflekshun** [reply]
Interesting, glad you enjoy the music and thank you SO much for finding this information, I'll run this by the dev I'm working with for sure. 

It seems unclear what the 'rate' values refer to. (0, 1, 2) seems to be the range. I could guess that 1 = uncompressed, 0.5 = half the sample rate (more compressed), but why would there then be 2 as a value (doubling audio rate doesn't increase quality after audio has been rendered). 

Still some questions, I will post any answers I find :) Will be cool to allow higher quality music in this seemingly great dev platform :)

Thanks again so much for the help so far!

**Reply by DigiEggz** [reply]
Sorry to ping, but did you ever figure out a solution to this issue? I've also noticed the audio sounding muffled in some browsers.

**Edit**: Opened a [PR](https://github.com/openfl/lime/pull/1788) to help resolve the issue.
